airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -600,7 +603,16 @@ private int getReturnCode(Pod pod) {
 
   @Override
   public int exitValue() {
-    return getReturnCode(podDefinition);
+    var returnCode = getReturnCode(podDefinition);
```

Comment:
It may be useful to have a comment to explain that `getReturnCode` is responsible for throwing the `IllegalThreadStateException` if the process hasn't completed yet.

Then it wouldn't look like calling `exitValue` always closes the process.

Reply:
good point

Reply:
done